wildlifelicensing/apps/applications/pdf.py

```diff
@@ -0,0 +1,182 @@
+from reportlab.lib.styles import getSampleStyleSheet, ParagraphStyle
+
+
+PAGE_WIDTH, PAGE_HEIGHT = A4
+
+DEFAULT_FONTNAME = 'Helvetica'
+BOLD_FONTNAME = 'Helvetica-Bold'
+ITALIC_FONTNAME = 'Helvetica-Oblique'
+BOLD_ITALIC_FONTNAME = 'Helvetica-BoldOblique'
+
+VERY_LARGE_FONTSIZE = 16
+LARGE_FONTSIZE = 14
+MEDIUM_FONTSIZE = 12
+SMALL_FONTSIZE = 10
+
+PARAGRAPH_INDENT = 15
+PARAGRAPH_BOTTOM_MARGIN = 2
+
+SECTION_BUFFER_HEIGHT = 10
+
+HEADER_MARGIN = 10
+
+PAGE_MARGIN = 20
+PAGE_TOP_MARGIN = 200
+
+DATE_FORMAT = '%d/%m/%Y'
+
+MISSING_VALUE_PLACEHOLDER = 'Not specified'
+MISSING_FILE_PLACEHOLDER = 'No file attachment provided'
+
+
+styles = getSampleStyleSheet()
+styles.add(ParagraphStyle(name='ApplicationTitle', fontName=BOLD_FONTNAME, fontSize=VERY_LARGE_FONTSIZE,
+                          spaceAfter=HEADER_MARGIN, alignment=enums.TA_CENTER))
+styles.add(ParagraphStyle(name='ApplicationVariantsTitle', fontName=BOLD_FONTNAME, fontSize=LARGE_FONTSIZE,
+                          spaceAfter=PARAGRAPH_BOTTOM_MARGIN * 2, alignment=enums.TA_CENTER))
+
+styles.add(ParagraphStyle(name='BoldLeft', fontName=BOLD_FONTNAME, fontSize=SMALL_FONTSIZE, alignment=enums.TA_LEFT))
+styles.add(ParagraphStyle(name='Left', alignment=enums.TA_LEFT))
+
+
+def _create_application(application_buffer, application):
+    every_page_frame = Frame(PAGE_MARGIN, PAGE_MARGIN, PAGE_WIDTH - 2 * PAGE_MARGIN, PAGE_HEIGHT - 2 * PAGE_MARGIN,
+                             id='EveryPagesFrame')
+    every_page_template = PageTemplate(id='EveryPages', frames=every_page_frame)
+
+    doc = BaseDocTemplate(application_buffer, pageTemplates=[every_page_template], pagesize=A4)
+
+    elements = []
+
+    elements.append(Paragraph(application.licence_type.name.encode('UTF-8'), styles['ApplicationTitle']))
+
+    # cannot use licence get_title_with_variants because licence isn't saved yet so can't get variants
+    if application.variants.exists():
+        variants = '({})'.format(' / '.join(application.variants.all().values_list('name', flat=True)))
+        elements.append(Paragraph(variants, styles['ApplicationVariantsTitle']))
+
+    elements.append(Spacer(1, SECTION_BUFFER_HEIGHT))
+
+    elements.append(_create_application_metadata(application))
+
+    elements.append(Spacer(1, SECTION_BUFFER_HEIGHT))
+
+    for field, datum in zip(application.licence_type.application_schema, application.data):
+        _create_application_questionaire(field, datum, elements, 0)
+
+    doc.build(elements)
+
+    return application_buffer
+
+
+def _create_application_metadata(application):
+    # dates and licensing officer
+    metadata_table_style = TableStyle([('VALIGN', (0, 0), (-1, -1), 'TOP')])
+
+    licensee = application.applicant.get_full_name()
+    if application.lodgement_number and application.lodgement_sequence:
+        lodgement_number = '{}-{}'.format(application.lodgement_number, application.lodgement_sequence)
```

Comment:
The `Application` model has a `reference` property for that. We might want to add the special case to it (no logment number or sequence).